packages/components/src/Combobox/Combobox.types.ts

```diff
@@ -17,3 +19,95 @@ export interface ComboboxOption {
   id: string | number;
   label: string;
 }
+
+export interface ComboboxProps {
+  readonly children: ReactElement | ReactElement[];
+
+  /**
+   * When `true`, `Combobox` will allow for multiple selections
+   *
+   * @default false
+   */
+  readonly multiSelect?: boolean;
+}
+
+export interface ComboboxActionProps {
+  /**
+   * The function to call when the action is clicked.
+   */
+  onClick(event: React.MouseEvent<HTMLButtonElement>): void;
+
+  /**
+   * The label text of the action.
+   */
+  readonly label: string;
+}
+
+interface ComboboxCloseProps {
+  /**
+   * Callback function invoked upon the selection of an option. Provides the selected option(s) as an argument.
+   */
+  readonly onSelect: (selection: ComboboxOption[]) => void;
+}
+
+interface ComboboxSelectProps {
+  /**
+   *
+   * Callback function invoked upon the Combobox menu closing. Provides the selected option(s) as an argument.
+   */
+  readonly onClose: (selection: ComboboxOption[]) => void;
+}
+
+interface ComboboxContentBaseProps {
+  /**
+   * List of selectable options to display.
+   */
+  readonly options: ComboboxOption[];
+
+  /**
+   * Optional action button(s) to display at the bottom of the list.
+   */
+  readonly children?: ReactElement | ReactElement[];
+
+  /**
+   * Placeholder text to display in the search input. Defaults to "Search".
+   */
+  readonly searchPlaceholder?: string;
+
+  /**
+   * pre selected option
+   * @default ""
+   * @type string
+   */
+  readonly selected: ComboboxOption[];
+
+  /**
+   * The encapsulating noun for the content of the combobox. Used
+   * in the empty state, and search placeholder. Should be pluralized.
+   */
+  readonly subjectNoun?: string;
+}
+
+export type ComboboxContentProps = ComboboxContentBaseProps &
+  XOR<ComboboxCloseProps, ComboboxSelectProps>;
+
+export interface ComboboxListProps {
+  readonly options: ComboboxOption[];
```

Comment:
so for this one in particular, no one _should_ be using ComboboxList. that's intended to be a completely internal piece of ComboboxContent that we moved into its own component with its own props purely to keep things atomic & focused

I'm not opposed to adding a bit more description to it, but it would only ever be people making changes to Combobox that would interact with it, never an end consume

Reply:
>  it would only ever be people making changes to Combobox 

Yeah I debated even commenting this one because it's not like they are external props - it's really just for anyone contributing to Combobox. I'm totally ok with leaving these as they are right now, for the sake of getting multi-select done 👍  Having the two other things updated makes me happy so don't worry about this one!